core/libdeno/libdeno_test.js

```diff
@@ -195,3 +195,65 @@ global.LibDenoEvalContextError = () => {
   assert(!errInfo5.isCompileError); // is NOT a compilation error! (just eval)
   assert(errInfo5.thrown.message === "Unexpected end of input");
 };
+
+global.LibDenoGCObserver = () => {
+  // Basic usage.
+  let num = 1;
+  const incr = () => {
+    Deno.core.print("Destroy callback invoked\n");
+    num = num + 1;
+  };
+  // IIFE. Simply using scope might not work.
+  (() => {
+    const o = Deno.core.setGCObserver({}, incr);
+    assert(!!o);
+  })();
+  // Guarantee GC.
+  gc();
+  assert(num === 2);
+
+  // Accessing the gc-ed object in callback.
+  let num2 = 0;
+  // IIFE. Simply using scope might not work.
+  (() => {
+    const obj = { a: 10 };
+    const incr2 = o => {
+      Deno.core.print("Destroy callback invoked 2\n");
+      num2 = o.a;
+    };
+    Deno.core.setGCObserver(obj, incr2);
+  })();
+  // Guarantee GC.
+  gc();
+  assert(num2 === 10);
```

Comment:
This one, using num2 seems like it could be a standalone test. It would be good to break up the test into distinct components so that they will fail individually during test_cc